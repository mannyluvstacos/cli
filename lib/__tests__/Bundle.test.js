const Application = require("@solfege/application");
const CommandsRegistry = require("../CommandsRegistry");
const Bundle = require("../Bundle");

describe("Bundle", () => {
  const command1 = {
    getName: () => {
      return "Command 1";
    }
  };
  const command2 = {
    getName: () => {
      return "Command 2";
    }
  };

  const runApplication = async (commands, parameters = []) => {
    const registry = new CommandsRegistry();
    commands.forEach(command => {
      registry.addCommand(command);
    });

    const application = new Application();
    application.setParameter("serviceContainer", {
      get: name => {
        if (name === "solfege_console_commands_registry") {
          return registry;
        }
      }
    });

    const bundle = new Bundle();
    application.addBundle(bundle);

    await application.start(parameters);
  };

  it("should display command list by default", async () => {
    let stdout = "";
    process.stdout.write = output => {
      stdout += output;
    };

    const commands = [command1, command2];

    await runApplication(commands);
    expect(stdout).toMatchSnapshot();
  });
});
