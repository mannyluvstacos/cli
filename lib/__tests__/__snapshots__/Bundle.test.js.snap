// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bundle should display command list by default 1`] = `
"[40m[36mSolfegeJS CLI[39m[49m
[40m[36m-------------[39m[49m

[32mCommand 1[39m   
[32mCommand 2[39m   
"
`;
